frontstage/views/sign_in/sign_in.py

```diff
@@ -67,10 +67,25 @@ def login():
             return render_template('sign-in/sign-in.html', form=form, data={"error": {"type": "failed"}})
         party_id = party_json['id']
 
-        oauth2_token = get_oauth_token(username, password, party_id, party_json, form, request.args.get('next'))
-
-        if not isinstance(oauth2_token, dict):
-            return oauth2_token
+        try:
+            oauth2_token = oauth_controller.sign_in(username, password)
+        except OAuth2Error as exc:
+            error_message = exc.oauth2_error
+            if USER_ACCOUNT_LOCKED in error_message:
+                logger.info('User account is locked on the OAuth2 server', party_id=party_id)
+                if party_json['status'] == 'ACTIVE' or party_json['status'] == 'CREATED':
+                    notify_party_and_respondent_account_locked(respondent_id=party_id,
+                                                               email_address=username,
+                                                               status='SUSPENDED')
+                return render_template('sign-in/sign-in.account-locked.html', form=form)
+            elif BAD_AUTH_ERROR in error_message:
+                return render_template('sign-in/sign-in.html', form=form, data={"error": {"type": "failed"}}, next=request.args.get('next'))
+            elif NOT_VERIFIED_ERROR in error_message:
+                logger.info('User account is not verified on the OAuth2 server')
+                return render_template('sign-in/sign-in.account-not-verified.html', party_id=party_id, email=username)
+            else:
+                logger.info('OAuth 2 server generated 401 which is not understood', oauth2_error=error_message)
```

Comment:
How do we know this generated a `401`?

Reply:
Because the controller gets to know the status code and spits out a special error (which isn't in the constants for some reason...).  I've tidied all of this up as it was kinda messy.